src/modules/Core/Context/reducer.js

```diff
@@ -1,25 +1,30 @@
 // @flow
 
 import type { DiskletFolder, EdgeContext } from 'edge-core-js'
+import { type Reducer } from 'redux'
 
-import type { Action } from '../../ReduxTypes'
+import { type Action, type Username } from '../../ReduxTypes.js'
+
+export type ContextState = {
+  context: EdgeContext,
+  folder: DiskletFolder,
+  usernames: Array<Username>,
+  nextUsername: Username
+}
 
 const initialState = {
   context: {},
   usernames: [],
   nextUsername: ''
 }
-export type State = {
-  context: EdgeContext | {},
-  usernames: Array<string>,
-  nextUsername: string
-}
-export const context = (state: State = initialState, action: Action) => {
+
+export const context: Reducer<ContextState, Action> = (state = initialState, action: Action) => {
```

Comment:
Reducer return type is completely broken.

Reply:
I don't know anything about this reducer. @wolverineks may be able to help.